src/ErrorCross/ErrorCross.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { findCorrectColor, initialColor } from '../utils'
+
+class Cross extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      color: initialColor,
+    }
+  }
+
+  componentDidMount() {
+    const newColor = findCorrectColor(this.div)
+    if (this.state.color !== newColor) {
+      this.setState({ color: newColor })
+    }
+  }
+
+  render() {
+    const { color } = this.state
+    const { style, className, message } = this.props
+
+    return (
+      <div
+        title={message}
+        ref={(c) => { this.div = c }}
```

Comment:
Now I prefer to do this : 
```es6
class extends Component {
  attach: () => {
    const newColor = findCorrectColor(this.div)
     if (this.state.color !== newColor) {
       this.setState({ color: newColor })
     }
  }

  render () {
    return <div ref={this.attach} />
  }
}
```

So I am sure of the order, and I don't rely on the `this` context